src/Sets/Zonotope.jl

```diff
@@ -371,19 +371,128 @@ nonlinear systems with uncertain parameters using conservative linearization.
 In Proc. of the 47th IEEE Conference on Decision and Control.*
 """
 function split(Z::Zonotope, j::Int)
-    @assert 1 <= j <= ngens(Z) "cannot split a zonotope with $(ngens(Z)) generators along index $j"
     c, G = Z.center, Z.generators
-    Gj = G[:, j]
-    Gj_half = Gj / 2
-
-    c₁ = c - Gj_half
-    c₂ = c + Gj_half
-
-    G₁ = copy(G)
-    G₁[:, j] = Gj_half
-    G₂ = copy(G₁)
 
+    c₁ = similar(c)
+    G₁ = similar(G)
     Z₁ = Zonotope(c₁, G₁)
+
+    c₂ = similar(c)
+    G₂ = similar(G)
     Z₂ = Zonotope(c₂, G₂)
+
+    split!(Z₁, Z₂, Z, j)
+end
+
+function split!(Z₁::Zonotope, Z₂::Zonotope, Z::Zonotope, j::Int)
+    c, G = Z.center, Z.generators
+    n, p = size(G)
+    @assert 1 <= j <= p "cannot split a zonotope with $p generators along index $j"
+
+    c₁, G₁ = Z₁.center, Z₁.generators
+    c₂, G₂ = Z₂.center, Z₂.generators
+    copyto!(G₁, G)
+    copyto!(G₂, G)
+
+    @inbounds for i in 1:n
+        α = G[i, j] / 2
+        c₁[i] = c[i] - α
+        c₂[i] = c[i] + α
+        G₁[i, j] = α
+        G₂[i, j] = α
+    end
+    return _split_ret(Z₁, Z₂)
+end
+
+_split_ret(Z₁::Zonotope, Z₂::Zonotope) = (Z₁, Z₂)
+
+function load_static_arrays()
+return quote
+
+function _split_ret(Z₁::Zonotope{N, SV, SM}, Z₂::Zonotope{N, SV, SM}) where {N, n, p, SV<:MVector{n, N}, SM<:MMatrix{n, p, N}}
+    Z₁ = Zonotope(SVector{n}(Z₁.center), SMatrix{n, p}(Z₁.generators))
+    Z₂ = Zonotope(SVector{n}(Z₂.center), SMatrix{n, p}(Z₂.generators))
     return Z₁, Z₂
 end
+
+end end  # quote / load_static_arrays
+
+"""
+    split(Z::Zonotope, gens::AbstractVector{Int}, n::AbstractVector{Int})
```

Comment:
Can we use a different name for `n`, which is usually a scalar for the dimension? Maybe `splits`?

Reply:
we could also use `partition`, `numsplits` , `nsplits`, .. this is a non-keyword arg though.